### ISSUE TITLE
Thermal Evaporation Towers Not Forming When Touching Other Multiblocks

### ISSUE THREAD

**Issue by DizzyDJW** [mekanism/Mekanism#8186]
### Issue description

Unlike the Boiler or Turbine or Fission Reactor, the Thermal Evaporation Tower will NOT form if touching another multiblock structure that is not like itself. I have one that is touching a Boiler and refuses to form, and another that is touching a Fission Reactor and refuses to form, however the ones directly adjacent to it, that are identical in every way, have formed successfully.

### Steps to reproduce

1. Build a Boiler, Turbine, or Fission Reactor Multiblock
2. Attempt to build Thermal Evaporation Tower touching said Multiblock.
3. See issue.

### Minecraft version

1.21 (Latest)

### NeoForge version

21.0.78-beta

### Mekanism version

10.6.7 (Latest)

**Comment by UnknownSerhan** [comment]
It is not recommended to build multiblocks touching each other at all, because of how multiblocks are coded. They sometimes try to share each other's data and cause data losses and other bad stuff.

**Comment by NullPointerCollection** [comment]
Please post a picture of what's not working. I'm guessing you're using glass to build the thermal plants and they're touching glass on the same side of another multiblock, which will cause this problem. If that's not the case, we'd still need to see pictures to see how exactly this issue is occurring.

**Comment by DizzyDJW** [comment]
> It is not recommended to build multiblocks touching each other at all, because of how multiblocks are coded. They sometimes try to share each other's data and cause data losses and other bad stuff.

Actually I read the exact opposite, that you SHOULD build them touching due to them not being able to function otherwise because the valves have no speed limits while no amount of pipes can keep up outside of creative pipes from other mods.

**Comment by DizzyDJW** [comment]
> Please post a picture of what's not working. I'm guessing you're using glass to build the thermal plants and they're touching glass on the same side of another multiblock, which will cause this problem. If that's not the case, we'd still need to see pictures to see how exactly this issue is occurring.

I'll try to get a picture, but no glass and actually I found that one Therm Evap Multiblock touching was fine but the moment a second therm Evap Multiblock touched the same Multiblock, be it a fission or turbine, that's when the issue occurs

**Comment by NullPointerCollection** [comment]
> > Please post a picture of what's not working. I'm guessing you're using glass to build the thermal plants and they're touching glass on the same side of another multiblock, which will cause this problem. If that's not the case, we'd still need to see pictures to see how exactly this issue is occurring.
> 
> I'll try to get a picture, but no glass and actually I found that one Therm Evap Multiblock touching was fine but the moment a second therm Evap Multiblock touched the same Multiblock, be it a fission or turbine, that's when the issue occurs

Does either the reactor or turbine have glass on that side with the TEPs?

**Comment by DizzyDJW** [comment]
> > > Please post a picture of what's not working. I'm guessing you're using glass to build the thermal plants and they're touching glass on the same side of another multiblock, which will cause this problem. If that's not the case, we'd still need to see pictures to see how exactly this issue is occurring.
> > 
> > 
> > I'll try to get a picture, but no glass and actually I found that one Therm Evap Multiblock touching was fine but the moment a second therm Evap Multiblock touched the same Multiblock, be it a fission or turbine, that's when the issue occurs
> 
> Does either the reactor or turbine have glass on that side with the TEPs?

Yes, both have glass, so the mekanism glass is the issue? Any chance this may someday be fixed, its fine if not, I can move the 8 TEP one block closer to the center and still fit all 8 in a single chunk with only a single block gap between each one instead of a 2 block gap. For aesthetics and more space in the center for Sodium production i was hoping to have more room.

**Comment by UnknownSerhan** [comment]
> > It is not recommended to build multiblocks touching each other at all, because of how multiblocks are coded. They sometimes try to share each other's data and cause data losses and other bad stuff.
> 
> Actually I read the exact opposite, that you SHOULD build them touching due to them not being able to function otherwise because the valves have no speed limits while no amount of pipes can keep up outside of creative pipes from other mods.

In general port to port is recommended because of transfer issues, yes, but not because it is safe to build them touching. Even port to port builds have a potential to fail because of them interacting in unwanted ways with each other. The best way is to completely stop their interaction is by using no glass that can be shared, or just don't build them touching at all unless you really want to do port to port, with TEPs you don't need to.

Also I don't know if I mentioned, but TEPs touching to same multiblocks will cause them to not form because again, they share glass. One way is to split the glass of the said multiblock you built the TEPs touching to via using casings. For example if you built a fission reactor and placed TEPs touching the reactor, place a single column of reactor casings so the reactor glass is split from each other. This stops the interaction between two TEPs.

**Comment by kienerxyzzy** [comment]
I don't know if this is related, but if you place ONE SINGULAR induction matrix casingnext to an already formed one, it stops being formed

**Comment by UnknownSerhan** [comment]
That's a basic multiblock behaviour, and isn't a bug.

**Comment by thiakil** [comment]
Due to how our multiblocks currently work, shared faces should not use structural glass (or other blocks of the same multiblock type) for this very reason.

It would likely require a major rewrite to change, so for the moment it is not high on the priority list to be fixed.
